src/elexmodel/handlers/data/Featurizer.py

```diff
@@ -33,27 +44,34 @@ def _center_features(self, df):
     def _add_intercept(self, df):
         df["intercept"] = 1
 
-    def _expand_fixed_effects(self, df, drop_first):
+    def _expand_fixed_effects(self, df: pd.DataFrame, drop_first: bool) -> pd.DataFrame:
         """
         Convert fixed effect columns into dummy variables.
         """
+        original_fixed_effect_columns = df[self.fixed_effect_cols]
+        # set non-included values to other as needed
+        fe_df = df.copy()
+        for fe, params in self.fixed_effect_params.items():
+            if "all" not in params:
+                fe_df[fe] = np.where(~fe_df[fe].isin(params), "other", fe_df[fe])
+
+        expanded_fixed_effects = pd.get_dummies(
+            fe_df, columns=self.fixed_effect_cols, prefix=self.fixed_effect_cols, prefix_sep="_", dtype=np.int64
+        )
+
+        # drop first column or "other" column if drop_first is true
+        cols_to_drop = []
+        if drop_first:
+            for fixed_effect in self.fixed_effect_cols:
+                relevant_cols = [col for col in expanded_fixed_effects.columns if col.startswith(fixed_effect)]
```

Comment:
the original fixed effect column is still in the `expanded_fixed_effects` dataframe, right? And the relevant columns shouldn't include the original fixed effect column, so should it not be `startswith(fixed_effect + '_')`?

Reply:
The original column gets dropped - that's why we have the concatenation of the two dataframes at the end (to preserve the original column)

Reply:
> I'm curious about the backward compatibility. Is there any particular reason to do that? Is it poor form not to?

Hmm I think it depends in terms of poor form. For our purposes, it probably doesn't matter. It was a combination of it being relatively easy to do + me being lazy and not wanting to update a million tests + also thinking that it wouldn't break people's test commands that they have. 

Do you think I should get rid of that? It does make things slightly more complicated but I tried to minimize that.

But this reminds me, I should update the docs at least.

Reply:
It also means we don't have to update the data importer calls but 🤞 we won't be using the model calls in data importer anymore....

Reply:
> The original column gets dropped - that's why we have the concatenation of the two dataframes at the end (to preserve the original column)

oh right, thanks!

Reply:
> > I'm curious about the backward compatibility. Is there any particular reason to do that? Is it poor form not to?
> 
> Hmm I think it depends in terms of poor form. For our purposes, it probably doesn't matter. It was a combination of it being relatively easy to do + me being lazy and not wanting to update a million tests + also thinking that it wouldn't break people's test commands that they have.
> 
> Do you think I should get rid of that? It does make things slightly more complicated but I tried to minimize that.
> 
> But this reminds me, I should update the docs at least.

I don't mind it. But I wonder if we should not have the dictionary allow for an `all` case then? And the old way just being the way of specifying all fixed effects? mostly I just don't love having multiple ways of doing the same thing, which I am worried might cause confusion. but I do not feel strongly about this, so happy to go with this now.

Reply:
That doesn't entirely work because because you can't submit multiple FE via the cli then. I did only specify the new way in the docs.

Reply:
ok